fendermint/actors/gas/src/lib.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2021-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use fil_actors_runtime::runtime::{ActorCode, Runtime};
+use fil_actors_runtime::ActorError;
+use fil_actors_runtime::SYSTEM_ACTOR_ADDR;
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::ipld_block::IpldBlock;
+use fvm_ipld_encoding::tuple::*;
+use fvm_shared::{MethodNum, METHOD_CONSTRUCTOR};
+use num_derive::FromPrimitive;
+
+#[cfg(feature = "fil-actor")]
+fil_actors_runtime::wasm_trampoline!(IPCEamActor);
+
+pub const IPC_GAS_ACTOR_NAME: &str = "gas";
+pub type Gas = u64;
+
+pub struct IPCGasActor;
+
+#[derive(Serialize_tuple, Deserialize_tuple, Debug, Clone)]
+pub struct State {
+    block_gas_limit: Gas,
+}
+
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Constructor = METHOD_CONSTRUCTOR,
+    SetBlockGasLimit = frc42_dispatch::method_hash!("SetBlockGasLimit"),
+}
+
+impl IPCGasActor {
+    /// Creates the actor
+    pub fn constructor(rt: &impl Runtime, args: ConstructorParams) -> Result<(), ActorError> {
+        let st = State {
+            block_gas_limit: args.block_gas_limit,
+        };
+        rt.create(&st)?;
+
+        Ok(())
+    }
+
+    fn set_block_gas_limit(rt: &impl Runtime, limit: Gas) -> Result<(), ActorError> {
+        if rt.message().caller() != SYSTEM_ACTOR_ADDR {
+            return Err(ActorError::forbidden("not system actor".into()));
+        }
+
+        rt.transaction(|st: &mut State, _rt| {
+            st.block_gas_limit = limit;
+            Ok(())
+        })?;
+
+        Ok(())
+    }
+}
+
+impl ActorCode for IPCGasActor {
```

Comment:
I'd call this EIP1559GasMarketActor if we're putting the EIP-1559 logic on chain.
We should define a common interface for gas market actors so that users can implement alternative gas markets and plug them in in their genesis.